packages/blade/src/components/ProgressBar/ProgressBar.tsx

```diff
@@ -0,0 +1,142 @@
+import styled from 'styled-components';
+import type { ReactElement } from 'react';
+import { ProgressBarFilled } from './ProgressBarFilled';
+import { FormLabel } from '~components/Form';
+import { makeAccessible, makeSize } from '~utils';
+import { Text } from '~components/Typography/Text';
+import { useId } from '~src/hooks/useId';
+import { useTheme } from '~components/BladeProvider';
+import Box from '~components/Box';
+import type { ColorContrastTypes, Feedback } from '~tokens/theme/theme';
+
+type ProgressBarCommonProps = {
+  accessibilityLabel?: string;
+  contrast?: ColorContrastTypes;
+  intent?: Feedback;
+  label?: string;
+  showPercentage?: boolean;
+  size?: 'small' | 'medium';
+  value?: number;
+};
+
+type ProgressBarVariant = 'progress' | 'meter';
+
+type ProgressBarProgressProps = ProgressBarCommonProps & {
+  variant?: Extract<ProgressBarVariant, 'progress'>;
+  isIndeterminate?: boolean;
+};
+
+type ProgressBarMeterProps = ProgressBarCommonProps & {
+  variant?: Extract<ProgressBarVariant, 'meter'>;
+  isIndeterminate?: undefined;
+};
+
+type ProgressBarProps<T> = T extends {
```

Comment:
Do we need conditional type here? 

```ts
type ProgressBarProps = ProgressBarProgressProps | ProgressBarMeterProps;

```

should work too right?

Reply:
It should. I did this initially but then changed it. Can't recall why. Anyway made the change now

Reply:
Looks good now, I think we did conditional type on the Text because we wanted to intelisense to properly show the fontWeight=regular|bold but here we don't need this.